JokeDAO/JokeDaoV2Dev
# feat: add rich text formatting

# Description

- Add rich text editor to send proposal form (handles the following: bold, italic, striked, code, blockquote, code block, link, bullet list, ordered list, horizontal rule, H1-H5, image, undo action, redo action) 

The rich text is implemented with the headless editor [Tiptap](https://tiptap.dev/). It allows us to customize not only the appearance of our rich text editor but also its functionalities.

## Type of change


# How Has This Been Tested?

1. Create a contest (or use this one: `/rinkeby/0xD1AEda717310cd1da027047DA9b135E634459394`
2. Go to your contest page
3. Click on "Submit proposal"
4. In the modal, instead of seeing a simple text area, you should now see a rich text editor

![Screenshot_20220822_100354](https://user-images.githubusercontent.com/15010369/185872881-7dd0fe42-dfda-470c-99f2-5b1803a395df.png)

5. Add your content (whatever will do)
6. Click on "toggle preview" : you should now see what the proposal would look like when posted. 
7. Click again on "toggle preview": you should now see the rich text editor again
8. Click on "Submit proposal"
9. After being submitted, if you click on "submit another proposal", the rich text should be empty
10. If you go back to the list of proposals, you should now see you rich proposal (cropped)
11. ![Screenshot_20220822_100412](https://user-images.githubusercontent.com/15010369/185873262-0093cab0-bcd5-4dda-97dd-8fa433b89ceb.png)
12. Click on the proposal you added
13. On the proposal page, you should see your full proposal
![Screenshot_20220822_100427](https://user-images.githubusercontent.com/15010369/185873424-fa1a7684-3842-44a9-99de-dc2e4d49512a.png)
14. Go to a previous contest (`/polygon/0x3aCD536121141BDA3E8C0910E21802DA5Be3fCae`) for instance
15. All the proposals should appear normally
